### PR TITLE
Add extraction process with ogr2ogr and fix creation of ZIP archive

### PR DESCRIPTION
**Pull request by Vampouille**
This PR also :
* introduce database in docker composition
* use gdal 2.1.0 and python 3.5 from debian stretch

### CODE REVIEW

**Comment by fvanderbiest** [comment]
you mean ogr2ogr ? ;-P

File: celery/extractions.py [hunk 1]
```diff
@@ -14,13 +20,109 @@
...
+
+def run_command(args):
+    """
+    Run command specified by args. If exit code is not 0 then full command line, STDOUT, STDERR are printed and an
+    Exception is raised
+    :param args: array of argument
+    :return: None
+    """
+    p = Popen(args, stdout=PIPE, stderr=PIPE)
+    p.wait()
```
Comment (fvanderbiest) [inline celery/extractions.py:38]: I'm wondering if we should specify a timeout argument to wait().
https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Comment (Vampouille) [reply celery/extractions.py:38]: yes good idea

Comment (Vampouille) [reply celery/extractions.py:38]: 10 min ?

File: celery/extractions.py [hunk 2]
```diff
@@ -14,13 +20,109 @@
...
+    :return: array of table name
+    """
+    cur = conn.cursor()
+    cur.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'foncier_%s'" % year)
+    res = [res[0] for res in cur.fetchall()]
+    cur.close()
+    return res
+
+
+def export_schema_to_shapefile_or_mapfile(year, proj, output_dir, format, conn, pg_connect_string):
```
Comment (fvanderbiest) [inline celery/extractions.py]: Please rename to `export_schema_to_shapefile_or_mapinfo` (no mapfile involved here ;-)

File: celery/extractions.py [hunk 3]
```diff
@@ -14,13 +20,109 @@
...
+
+def export_schema_to_sql(year, proj, output_dir, conn, pg_connect_string):
+    """
+    Extract all table in schema foncier_<year> where 'year' is the first argument and one sql file to create schema,
+    tables and inserts data
+    :param year: numeric year to append to 'foncier_' to build schema name
+    :param proj: output projection
+    :param output_dir: folder where files will be writen
+    :param conn: a psycopg connection instance
+    :param pg_connect_string: a string that contains option to connect to database with ogr2ogr. 'schema' option will
```
Comment (fvanderbiest) [inline celery/extractions.py]: schemas, not schema

File: celery/extractions.py [hunk 4]
```diff
@@ -14,13 +20,109 @@
...
+    be added
+    :return: None
+    """
+    with open(join(output_dir, "foncier_%s.sql"  % year), 'wb') as f:
```
Comment (fvanderbiest) [inline celery/extractions.py]: `wb` ? no binary involved here, just plain text, no ?

File: celery/extractions.py [hunk 5]
```diff
@@ -14,13 +20,109 @@
...
+            with open(table_output_file, 'rb') as table_file:
+                f.write(table_file.read())
+            remove(table_output_file)
+
+
 @taskmanager.task(name='extraction.do')
 def do(year, format, proj, email, cities):
-    tmpdir = tempfile.mkdtemp(dir = FONCIER_EXTRACTS_DIR, prefix = 'foncier_{0}_{1}_{2}_{3}-'.format(year, format, proj, do.request.id))
+
+    extraction_id = 'foncier_{0}_{1}_{2}_{3}'.format(year, format, proj, do.request.id)
```
Comment (fvanderbiest) [inline celery/extractions.py]: `foncier_{0}_{1}_{2}_{3}-` is nicer IMO, but no blocker.

File: celery/extractions.py (hunk 1 above)
Comment (fvanderbiest) [reply celery/extractions.py:38]: No idea. 
I guess a new ENV var for this would allow us to be more flexible.

**Comment by fvanderbiest** [comment]
Finished my review, LGTM.
Thanks for the great code.

File: celery/extractions.py (hunk 4 above)
Comment (Vampouille) [reply celery/extractions.py]: There are some UTF-8 characters that are easier to process with binary flag. I have to add this flag after testing against test data.

File: docker-compose.yml [hunk 6]
```diff
@@ -20,11 +20,13 @@ worker:
   environment:
     - CELERY_BROKER_URL=redis://redis:6379/0
     - CELERY_RESULT_BACKEND=redis://redis:6379/0
+    - PG_CONNECT_STRING=host=database port=5432 user=geoserver dbname=geoserver password=geoserver
```
Comment (fvanderbiest) [inline docker-compose.yml:23]: `sslmode=require` ? I'm not sure about it.

Comment (Vampouille) [reply docker-compose.yml:23]: not needed for docker setup

File: celery/extractions.py (hunk 5 above)
Comment (Vampouille) [reply celery/extractions.py]: Ok, I will use this format.

**Comment by fvanderbiest** [comment]
Oops, I completely forgot that we should filter the results on the cities list !
Maybe you planned it for a subsequent PR ?

**Comment by Vampouille** [comment]
> Maybe you planned it for a subsequent PR ?

Yes

**Pull request merged**